### PR TITLE
KAFKA-13982: Move WorkerConfigTransformerTest to use Mockito

### PR DESCRIPTION
**Pull request by clolov**
Addressing https://issues.apache.org/jira/browse/KAFKA-13982.

### CODE REVIEW

**Comment by clolov** [comment]
@C0urante I believe you might have the required context to review this pull request :)

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java
Comment (C0urante) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java:150]: Nice catch, took me a second to figure out the misalignment between the type signatures for this method and for `ConfigProvider::unsubscribe`.

Should we also remove the `subscribe` method, since it matches the [default implementation](https://github.com/apache/kafka/blob/693e283802590b724ef441d5bf7acb6eeced91c5/clients/src/main/java/org/apache/kafka/common/config/provider/ConfigProvider.java#L59-L61) for `ConfigProvider::subscribe`?

Also, should we annotate any remaining methods with `@Override` to help prevent future mistakes like the one you've caught here?

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java
Comment (C0urante) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java]: Why not use the Mockito `@Mock` annotation here?

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java
Comment (C0urante) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java:89]: Good catch 👍

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java
Comment (C0urante) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java:101]: Should we add a verification here that `herder::restartConnector` was invoked?

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java
Comment (C0urante) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java:125]: Same thought here RE verifying `herder::restartConnector`

**Review by C0urante — COMMENTED** [review]
Thanks @clolov, some good finds here!

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java
Comment (clolov) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java]: If I use `@Mock` I get the following error in some of the tests
```
java.lang.NullPointerException: Cannot invoke "org.apache.kafka.connect.runtime.Worker.herder()" because "this.worker" is null
```
I saw that there are multiple approaches to solve this https://stackoverflow.com/questions/29590621/mock-instance-is-null-after-mock-annotation. Are we following any of those in the code base? If so, I will just change to that approach.

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java
Comment (clolov) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java:150]: Agreed and added.

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java
Comment (clolov) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java:101]: Added

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java
Comment (clolov) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java:125]: Added

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java
Comment (clolov) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java]: I decided to go for `@RunWith(MockitoJUnitRunner.class)` in the newest commit, but I am happy to change it.

Comment (ijuma) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java]: Since we are in the process of migrating to JUnit 5, it's simpler if we don't rely on JUnit 4 specific features. It's fine to use the `mock` method, but please use static imports to make it more concise.

Comment (C0urante) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java]: It's pretty straightforward to replace `@RunWith(MockitoJUnitRunner.class)` with `@ExtendWith(MockitoExtension.class)`, and we're already using that approach in other modules (such as [here](https://github.com/apache/kafka/blob/badfbacdd09a9ee8821847f4b28d98625f354ed7/core/src/test/java/kafka/test/ClusterTestExtensionsTest.java#L36)). We also get to leverage things like the [`@MockitoSettings` annotation](https://www.javadoc.io/doc/org.mockito/mockito-junit-jupiter/2.19.1/org/mockito/junit/jupiter/MockitoSettings.html) if we stick to this style.

@ijuma since it's only going to be a one-line (well, excluding imports) change per test, do you think we could stick with `@RunWith` for now in our Mockito tests?

**Review by C0urante — APPROVED** [review]
LGTM modulo the open item about the `@RunWith` annotation, thanks @clolov!

**Comment by clolov** [comment]
Heya @C0urante and @ijuma I just want to confirm whether we are waiting on a change which I need to make?

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java
Comment (ijuma) [reply connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTransformerTest.java]: Yeah, that's fine.

**Comment by ijuma** [comment]
@C0urante you can use your committer privileges to merge this if you're happy with it (and the test results look good). :)

**Comment by C0urante** [comment]
Thanks @ijuma. No test failures in Connect and all others appear unrelated; this looks good to go.

Thanks for the PR @clolov!

**Pull request merged**
